deploy/sre-prometheus/100-sre-api-errors.PrometheusRule.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: k8s
+    role: alert-rules
+  name: sre-uptime-sla
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-uptime-sla
+    rules:
+    - alert: SLAUptimeSRE
+      annotations:
+        message: The API server has had 100 percent request failures for 300 seconds.
+      expr: |
+        (sum(rate(apiserver_request_count{job="apiserver", code=~"(400|5..)"}[5m])) / sum(rate(apiserver_request_count{job="apiserver"}[5m])) >= 1)
```

Comment:
@c-e-brumm I don't think this aligns with the way we are measuring uptime.  Could you comment on this with respect to the language in the contract appendix?

Reply:
For reference this is what it says in the appendix:
`Monthly Availability = 
100 * [(Total number of 5 minute intervals in a calendar month - Number of Downtime intervals) /  (Total number of 5 minute intervals in a calendar month)] ` 

https://docs.google.com/document/d/1Qa3tVePF94ZzrThMyvqNySwLwtQGxVshqIHS8D_N3XQ/edit?skip_itp2_check=true&pli=1#

Reply:
@jewzaam After some discussion we went back to measuring by 5 minute increments with no extra math. This alert is part of measuring that.

Reply:
Synced w/ @c-e-brumm off PR.  Summary is we do expect to report on outages less than 5 minutes and using `rate(...[5m])` does not meet that need.

Reply:
@jewzaam 

From the text in the link:

```“Monthly Availability” means the total number of five minute intervals in a calendar month, subtracting the number of Downtime intervals during the calendar month and dividing that number by the total number of five minute intervals in the calendar month.  ```

Where a Downtime interval is defined as:
```“Downtime” means a period of more than five minutes in which all connection requests to the OpenShift Dedicated Cluster’s API server fails. ```

Could you please clarify a little more as I can't find anywhere that we must report on outages less then 5 minutes.
Happy to have a sync off-PR too